core.html

```diff
@@ -42,70 +34,90 @@ <h2>Method Specific Identifier</h2>
 
     <p>The peer DID scheme is defined by the following ABNF (see [[RFC5234]] for syntax):</p>
     <pre class="example" class="ABNF" title="ABNF for peer DIDs">
-peer-did = "did:peer:" numalgo encalgo "-" numbasis
+peer-did = "did:peer:" numalgo transform encnumbasis
 numalgo = "1"
-encalgo = "1"
-numbasis = 64*HEXDIGCI
-HEXDIGCI = HEXDIG / "a" / "b" / "c" / "d" / "e" / "f"
+transform = "z"
+encnumbasis = 45*BASE58
```

Comment:
Why 45 bytes? Maybe I'm misremembering the ed25519 public key size... thought it was 256 bits, which can be encoded in base58 using 22 characters. Doing this from memory, so I'm probably wrong.

Reply:
256 bits = 32 bytes when it's binary, or more for non-binary encodings. A base58 encoding of a 32-byte value is 44 ASCII characters. However, when you factor in our need to also encode the multicodec identifier 0x12, we need another ASCII character. (Actually, now that I think about it, we might need enough bits for 2 extra characters. I'll check...)

You may be remembering that Sovrin's DIDs have 22 characters after the prefix. That's because Sovrin only uses the upper 128 bits of the public key as the DID value--not the full 256 bits.

Reply:
> 256 bits = 32 bytes when it's binary, or more for non-binary encodings. A base58 encoding of a 32-byte value is 44 ASCII characters.

You're right, I was doing the math incorrectly in my head and getting further confused when I did the hex encoding... :)

> encnumbasis = 45*BASE58

minor nit, but you'll want to say BASE58BTC (since 'z' is the Bitcoin alphabet encoding of base58)

Reply:
I've carefully examined my assumptions and the math. With the 0x20 digest length in multicodec, we have 34 bytes or 2^272 bits to encode. This is 7.5885e81. This would take 47 Base58 characters to encode (58^46 = 1.31e81, and 58^47=7.6e82). However, since we know that the first byte of the 34 bytes will be 0x12, we know that we don't need the full expressivity of 58^47. It turns out that 0x12-followed-by-33-other-bytes can always be encoded in 46 Base58 chars, so I've updated the spec's ABNF, narration, and regex to call for 46 Base58 chars.